app/policies/result_policy.rb

```diff
@@ -44,7 +44,7 @@ def grade?
   end
 
   def review?
-    role.instructor? || role.ta? || (
+    role.instructor? || (role.ta? && record.grouping.tas.exists?(role.id)) || (
```

Comment:
In addition to the above, I'd like you to move the actual "assigned TA" check into the `GroupingPolicy` class, similar to the existing `member?` policy there.

You might call it something like `assigned_grader?`, so that in here you can call `check?(:assigned_grader?, ...)`, similar to how other policies in this file use `member?` as a "helper" policy.

This will help us re-use this policy check in other methods (see my overall review comment).